include/robot_interfaces/robot.hpp

```diff
@@ -0,0 +1,86 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2019, Max Planck Gesellschaft
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <robot_interfaces/robot_frontend.hpp>
+#include <robot_interfaces/robot_backend.hpp>
+#include <robot_interfaces/robot_data.hpp>
+#include <type_traits>
+
+namespace robot_interfaces
+{
+
+/**
+ * @brief RobotFrontend that construct and encapsulates
+ * its related RobotBackend. It also construct and starts
+ * the robot driver.
+ */
+template<typename Action,
+	 typename Observation,
+	 typename Driver,
+	 typename ... Args>
```

Comment:
It might be good extending the demo a bit to make use of this feature.

Reply:
@luator 
Not sure what you mean, I also updated the demo to perform the two version (frontend/backend, and robot)

Reply:
I mean the arguments for the driver (the `... Args` part).  In the demo, the driver does not take any arguments.  I am not familiar with this syntax, I assume it would work like the following?

    robot_interfaces::Robot<Action, Observation, Driver, int, double> robot(max_duration, max_inter, 42, 3.14);

Reply:
Not intuitively, it would work as:

robot_interfaces::Robot<Action, Observation, Driver> robot(max_duration, max_inter, 42, 3.14);

+1 on updating the demo, will do

Reply:
done